src/SFML/Graphics/Shader.cpp

```diff
@@ -868,89 +868,49 @@ bool Shader::compile(std::string_view vertexShaderCode, std::string_view geometr
     // Create the program
     const GLEXT_GLhandle shaderProgram = glCheck(GLEXT_glCreateProgramObject());
 
-    // Create the vertex shader if needed
-    if (!vertexShaderCode.empty())
+    // Helper function for shader creation
+    const auto createAndAttachShader = [&](GLenum shaderType, const char* shaderTypeStr, std::string_view shaderCode)
```

Comment:
```suggestion
    const auto createAndAttachShader = [shaderProgram](GLenum shaderType, const char* shaderTypeStr, std::string_view shaderCode)
```
If a lambda only captures 1 or 2 things I'd prefer we explicitly capture those variables.

Reply:
Good point :)